__tests__/app.test.js

```diff
@@ -99,68 +99,183 @@ describe("/api/articles/:article_id", () => {
-                expect(article.hasOwnProperty("comment_count"))
-            })
+            const newComment = body
+            expect(typeof newComment).toBe('object')
+            expect(Array.isArray(newComment)).toBe(false)
+            expect(newComment.author).toBe("icellusedkars")
+            expect(newComment.body).toBe('What an incredible article. I am in awe of this.')
         })
     })
-    test("GET: 200 articles are sorted by created_at in descending order", () => {
+    test("POST 201: the new comment object has the same properties that all the other comments have", () => {
         return request(app)
-        .get("/api/articles")
-        .expect(200)
+        .post("/api/articles/1/comments")
+        .send({
+            username: "icellusedkars",
+            body: "What an incredible article. I am in awe of this."
+        })
+        .expect(201)
         .then(({ body }) => {
-            expect(body).toBeSortedBy("created_at", {
-                descending: true
-            })
+            expect(body.hasOwnProperty("comment_id"))
+            expect(body.hasOwnProperty("body"))
+            expect(body.hasOwnProperty("author"))
+            expect(body.hasOwnProperty("article_id"))
+            expect(body.hasOwnProperty("created_at"))
+            expect(body.hasOwnProperty("votes"))
         })
     })
-    test("GET: 200 article objects have a comment_count property that is equal to the number of comments associated to that article", () => {
+    test("POST 400: responds with appropriate error message when the article_id provided is not a number", () => {
         return request(app)
-        .get("/api/articles")
-        .expect(200)
+        .post("/api/articles/article_one/comments")
+        .send({
+            username: "icellusedkars",
+            body: "What an incredible article. I am in awe of this."
+        })
+        .expect(400)
         .then(({ body }) => {
-            
-            body.forEach((article) => {
-                expect(article.hasOwnProperty("comment_count"))
-            })
-            expect(body[0].comment_count).toBe(2)
-            expect(body[6].comment_count).toBe(11)
+            expect(body.msg).toBe('Bad request')
         })
     })
-    test("404: responds with an correct error message if the path is invalid", () => {
+    test("POST 404: responds with appropriate error message when article_id is that of an article that doesnt exist", () => {
         return request(app)
-          .get("/api/article")
+          .post("/api/articles/100/comments")
+          .send({
+            username: "icellusedkars",
+            body: "What an incredible article. I am in awe of this."
+        })
           .expect(404)
           .then(({ body }) => {
-            expect(body.msg).toBe("path not found");
+            expect(body.msg).toBe("Article not found");
           });
       });
-})
\ No newline at end of file
+})
+
```

Comment:
We'll need a couple more tests.  
1. a 400 for when the comment body is not in the right format. 
2. Status 404, username does not exist
3. A 201 for when they're are unnecessary extra properties on the request body (they're just ignored).